### PR TITLE
Finally added proper support for multiple blindfolded!

### PR DESCRIPTION
**Pull request by jfly**
This is the last part of https://github.com/thewca/worldcubeassociation.org/issues/1922, which brings TNoodle up to date with the latest WCIF spec with regard to 333mbf scrambles. You can see the relevant WCIF documentation [here](https://docs.google.com/document/d/1hnzAZizTH0XyGkSYe-PxFL5xpKVWl_cvSdTzlT_kAs8/edit#heading=h.oaxof956lc4g).

Each scramble is now actually a newline separated list of scrambles.
This allows each scramble (as far as TNoodle is concerned) to correspond
to each attempt in the round.

**You can download a compiled version of TNoodle with these changes [here](https://www.dropbox.com/s/6e3u3qg32fmwk2f/TNoodle-WCA-0.11.5-9-ge909785.jar?dl=1). Please try it out and give feedback!**

## Screenshot of the new UI

I changed "# Scrambles" to "# Attempts". The "# Extra Scrambles" column now lets the user select how many puzzles they expect people to solve in each attempt. It defaults to 60, which is hopefully crazy high enough that it should be enough for everyone =) **I'm definitely open to changing that number to something else.**
 
image

## Screenshot of the generated PDFs

image

image

## Note that the structure of the zip file remains the same

```
~/tmp/test @kaladin> unzip -l Jfly\ mbf\ test.zip 
Archive:  Jfly mbf test.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
    41615  2017-09-08 20:41   pdf/3x3x3 Multiple Blindfolded Round 1 Group A.pdf
     1187  2017-09-08 20:41   txt/3x3x3 Multiple Blindfolded Round 1 Group A.txt
    41632  2017-09-08 20:41   pdf/3x3x3 Multiple Blindfolded Round 1 Group B.pdf
     1191  2017-09-08 20:41   txt/3x3x3 Multiple Blindfolded Round 1 Group B.txt
    83964  2017-09-08 20:41   pdf/3x3x3 Multiple Blindfolded Round 2 Group A.pdf
     2698  2017-09-08 20:41   txt/3x3x3 Multiple Blindfolded Round 2 Group A.txt
    83937  2017-09-08 20:41   pdf/3x3x3 Multiple Blindfolded Round 2 Group B.pdf
     2656  2017-09-08 20:41   txt/3x3x3 Multiple Blindfolded Round 2 Group B.txt
     8946  2017-09-08 20:41   Jfly mbf test.json
     8968  2017-09-08 20:41   Jfly mbf test.jsonp
      539  2017-09-08 20:41   Jfly mbf test.html
   231848  2017-09-08 20:41   Jfly mbf test.pdf
---------                     -------
   509181                     12 files
```

## Example of the new JSON

<details>

```json
~/tmp/test @kaladin> cat Jfly\ mbf\ test.json | jq
{
  "sheets": [
    {
      "scrambles": [
        "R2 D' B2 D' F2 R2 F2 L2 D' U F L' F2 R2 B R F' D' B2 D' Fw Uw2\nL2 U F2 D' L2 U' R2 D' U2 B F' U B' D2 F2 R' D' U2 L' F' Rw\nF2 L2 R2 D' R2 F2 R2 U' F' D F L' B' U' F L U2 R' D' Fw' Uw2\nB2 R' U2 B2 L2 R F2 D2 L D2 F' D2 R' U L2 D B2 F2 U Fw'\nL2 F2 U L2 D L2 U' R2 D R2 F' D' L U' R' D F2 D2 B L2 F Uw\nU F2 D B2 U2 F2 R2 D R U' L2 R2 F2 R2 F' L2 D' R' F R2 Fw\nR2 B F2 R2 B2 F' D2 U' L' B2 U L' R' B U B' F U F Rw2\nR2 U2 R' D2 U2 B2 L D2 L' B' L2 U' R D2 R' B' F' L U B2 Uw\nU2 F2 L2 B2 F' U2 L2 B R2 U' L D2 U R' B2 F U L' R2 U Rw2 Uw'\nU2 B R2 B' D2 B' L2 D2 U2 L' U B2 F' U' F2 L' F2 L' D' U Fw' Uw'",
        "F2 U2 F2 R2 U2 L2 D2 U B2 F' U' B2 D2 L F' R' B2 U B' R Fw\nF2 R2 D L2 U2 L2 F2 D2 U' B D F L' R' F2 L B R' U2 Rw' Uw\nR2 B2 F U2 L2 F' R2 U L' B' R F L' U2 B2 L D R' U Rw' Uw'\nU L2 F2 U' F2 L2 R2 D L2 R D2 B U2 R U' B2 L' B' U F2 Uw2\nR D2 L R2 U2 F2 L F2 L2 D L' B2 R2 F D2 B' D' R' B2 L2 Uw\nB2 L2 D2 L2 U' L2 R2 U' L2 B' D2 L2 F' U L' B2 F2 D L Fw\nL' R2 B2 R' B2 U2 F2 L2 D2 U R2 D2 U L2 R' F' D' B' D2 R Uw2\nD' B2 L2 B2 D2 L2 D U2 F L2 D' F' D' B L U' B' U R' B Rw Uw\nD2 L2 D' U2 R2 U R2 B2 F' L' U R F U' R2 B U R2 U2 R' Fw Uw\nR2 B U2 B D2 B D2 F U2 F R2 D' R U B F D' R' U' F' U2 Rw Uw2"
      ],
      "scrambler": "333ni",
      "copies": 1,
      "title": "3x3x3: Multiple Blindfolded Round 1 Group A",
      "fmc": false,
      "group": "A",
      "event": "333mbf",
      "round": 1
    },
    {
      "scrambles": [
        "F2 L2 F' D2 U2 F' D2 B' F R U2 R B F D' R' D L' F2 D Rw' Uw2\nU L2 D' R2 D' B2 L2 R2 D' L D2 L2 D' B L B' F U' R2 U2 Rw Uw2\nF L2 R2 B' D2 F D2 R' B L D F L2 D U2 R' U2 B2 D' Fw\nU R2 F2 D2 R2 D U2 F2 U2 R F U R' B2 D2 B2 R' F' D' U' Fw Uw'\nU2 R2 U F2 U' L2 B2 U' F2 D' R B' U' B D L' D2 U2 L' U' F' Rw'\nB2 L' D2 L D2 U2 R' U2 B2 U2 B F D' L B' U2 R F U' F2 Rw Uw'\nF' U2 B' L2 B' F L' F' D B L F R2 D' B2 L' B2 F' Rw\nL2 F2 D2 U' F2 D B' L2 U' F' L' F D B U2 F' U F' Uw\nL2 F R2 F2 R2 U2 B' R2 F2 R2 D' L R2 U' R2 U R2 D2 F R Fw' Uw2\nR2 B2 U2 L2 F D2 R2 F2 U' F R' F R2 B' R2 B' U' B R Fw Uw2",
        "D F2 R2 U R2 U' F2 D L' R' B' D' L' F' L2 F U2 F' D2 U Fw\nF2 U2 R' U2 L2 R' F2 R2 B2 F' D' L F2 L2 D2 B F' U' B2 R' Fw' Uw\nD2 L2 B2 D' L2 R2 B2 U' R2 D B2 F R' F2 D' U' R2 B2 U' R2 F' Rw Uw\nD2 R2 U2 F2 R2 U R2 D' B2 U2 R B U2 F2 R2 U' B2 L2 R' U' Rw2 Uw\nF2 L' U2 R2 U2 F2 L F2 D2 L' B2 D' U' B L D2 R2 B' F2 L B2 Rw Uw\nD' B2 D F2 U2 B2 R2 D2 R' F2 R' B R D2 B R2 D2 F2 D' B' Rw' Uw2\nD' L2 U2 R2 D' U2 R2 U R' U' B' U L U2 F2 U' F2 L U2 Fw Uw\nU2 B2 R' U2 F2 U2 L' R F L U R2 D L' U B' R D Rw\nR2 D R2 F2 D2 F2 U' B2 U' B' L D F2 R D2 B' R F U2 L Uw2\nU B2 U L2 D' B2 D' F2 L2 R' B D2 F' R' D F U B L U Fw'"
      ],
      "scrambler": "333ni",
      "copies": 1,
      "title": "3x3x3: Multiple Blindfolded Round 1 Group B",
      "fmc": false,
      "group": "B",
      "event": "333mbf",
      "round": 1
    },
    {
      "scrambles": [
        "U2 F L2 D2 B R2 B2 U2 L' D R2 U2 B F2 R' F' L2 F R2 Fw' Uw2\nF2 L' D2 L2 B2 D2 F2 D2 L' B' F L U2 B L2 D2 U L2 R Fw\nL2 D2 B2 R U2 L D2 L' R D U' B D' F' R' D2 F L R Fw' Uw\nD2 L' D2 U2 R B2 R2 D2 L' B2 R' D B' U F2 R' D B2 F' D F Uw'\nL2 U2 R' U2 B2 R2 F2 L F2 D' F2 U' F2 U B' R D' F' U F2 Rw2 Uw'\nD2 F2 U F2 R2 B2 R2 D' F L' D' F D2 F' R2 F L' D2 F' U Fw' Uw\nR2 B2 D' B2 D L2 F2 U' F U2 L' B' U2 L' D L2 D' F2 L2 Fw Uw\nL2 B2 D2 L' B2 R F2 L D2 F L R' F2 L' F' D R B F2 Rw2 Uw'\nR2 D2 F2 R2 F U2 L2 D2 F' L' B R' U' B2 F D' B2 D2 B' F U2 Fw Uw2\nR2 B2 D B2 L2 R2 D U2 R B' R2 F U L2 F R2 D2 L F' U Fw'\nU2 B R2 U2 F2 R2 D2 B L2 U' L2 B' F' R2 D' L D' U' L' B' Rw' Uw\nF' L2 D2 L2 D2 F D2 B D2 R D F2 D' R' D' B2 D B Rw\nU2 L2 B2 R2 U L2 D' R2 U2 B2 R' F' L' F R2 F2 D U R' U R2 Fw Uw2\nB' L2 R2 U2 R2 B F D2 B' R F D R' F2 U R D F U F' Rw2 Uw2\nR' F2 L2 U2 L' U2 L D2 R' F' R' B' L R' D' U L D' Rw2 Uw2",
        "U' F2 L' F' D R B' L2 F L' U2 R2 L2 F2 U F2 B2 U' R2 U'\nU2 B2 R' U2 B2 L' F2 R2 B2 R2 F' R2 D' U' B' L' B' R' U2 L2 F' Rw Uw\nU2 L' R D2 F2 R' D2 U2 L' R2 F' D2 B' U L' D B2 L R B' U' Rw' Uw2\nR2 U2 R' D2 U2 B2 L2 U2 F' R B2 U' L2 F2 R' F' U2 B L' Uw\nD L2 D B2 L2 U' B2 D R' D' F2 U2 F R B2 L2 D2 F' R2 B' Rw' Uw'\nF2 U F2 U' B2 L2 R2 D' B2 R2 B2 R U' L B' R D R2 D F' U' Rw' Uw\nF2 U2 R2 F L2 B2 R2 D2 F' U2 B' L' F L2 U2 L' U' L2 D B R' Fw Uw\nD2 B2 R2 U' B2 U R2 B2 F' R2 F2 R U L U R2 F L2 U' B2 Rw'\nU2 F' U2 L2 D2 F2 D2 B F2 D2 U' R' B2 R F' U B L U' F' Uw'\nD' L2 B2 L2 D' B2 U' R2 F' R2 U2 F' R B' F' D' R U' B2 F2 Rw' Uw'\nU2 L2 B' R2 F' D2 F U2 F2 U' B D' B' F' U' R' B L' U R' Fw Uw2\nD' L2 U2 L2 R2 D' R2 F2 D2 L' U B' F2 D' B' R F' L2 B' Rw Uw\nF2 U2 F2 R D2 L2 R' D2 L2 B L D2 L2 B' D B D2 F L R Fw Uw2\nD F' R D F' B' L U' D B D2 F2 U2 R' D2 R F2 R2 L\nB2 U2 B2 D' L2 B2 D' R2 D2 L2 B F R' B' D2 B2 L U L' F2 D2 Rw Uw",
        "D2 L F2 L2 R F2 R2 D2 R' B' U F D' U' L R2 B' F' Uw2\nL B2 L F2 L2 F2 L F' R D F L2 B' L' F2 L D2 L Fw' Uw\nL2 D2 U' F2 R2 D F2 R2 F2 D' U' L' F' U2 L' B2 U2 B' R2 F' D Rw Uw'\nD L2 F2 R2 B2 U2 F2 U B' R2 D B R F' L2 D' U' B R' U2 Rw' Uw2\nU' B2 U B2 R2 F2 U2 F D' F2 L2 D' B F2 L' D L B R' Fw'\nB D2 F D2 L2 B' F2 D2 R2 D2 R' U' F' L R2 D' B2 L' D' B' Uw2\nB2 D R2 D2 L2 B2 D B2 F R' B' F2 D R D R2 F' U' R D' Fw Uw\nL D2 U2 L' U2 B2 L2 B2 R' B' R2 B2 U' L U B' R2 D U' F Rw2 Uw2\nR2 B2 L' D2 L' D2 B2 D2 L' B F2 L2 U' B' D F U L2 R D' Rw Uw\nR' U2 L2 U2 B2 R D2 L B2 D2 L' F' U' B' D2 F' L2 D' L2 Fw' Uw\nF2 D F2 L2 D L2 D2 U' R B R2 B F R' D U2 R' U L Fw Uw\nD F2 D B2 R2 U2 L2 U' F2 R B2 F2 D2 R2 U F R' D' L U' Fw' Uw'\nB2 F2 R U2 L B2 D2 B2 L2 U2 R2 D U' F D U2 B F' D U2 Fw'\nF2 R2 B2 F2 U F2 D R2 B2 R B U L U2 R F2 R2 F' D2 U Rw2 Uw'\nR2 D' F2 D B2 U B F' R2 U' R F' U2 F2 L2 F L F' Rw2"
      ],
      "scrambler": "333ni",
      "copies": 1,
      "title": "3x3x3: Multiple Blindfolded Round 2 Group A",
      "fmc": false,
      "group": "A",
      "event": "333mbf",
      "round": 2
    },
    {
      "scrambles": [
        "D2 R2 F' L2 R2 U2 B L2 B' D U2 L B2 D U' B R B U2 R2 Fw\nU' B2 L2 D L2 D2 F2 D' F2 D F R D R2 B L' B2 F' L' B2 U Rw\nR2 F L2 U2 R2 F' L2 F' R U R2 B2 F' L R2 D' L B' L' Fw' Uw'\nL R F2 L B2 L' F2 R' F' D' U' B' L D2 R' F' U F Uw'\nL2 D2 U' L2 D2 U2 F D' F2 D2 U' L U' R' B U' R Fw Uw2\nD2 B L2 R2 F' D2 B R D2 B D' U' F2 L2 D' F D' L2 B' Rw' Uw\nR2 F2 D2 B2 R2 F2 R' F2 L U' F R B' L D' R' B2 D' R' D' Fw\nB2 D2 B2 L2 U2 R2 D2 L U2 R B U' L F D2 R' D B2 L' F Rw Uw2\nL2 F' L2 B2 R2 B2 U2 F2 U L U2 L2 U2 F L2 D B2 D2 B Rw Uw2\nR2 B2 F2 D L2 B2 F2 U' F2 R B' F' L U' B' R B' F D' B2 Rw Uw2\nB2 U' B2 R2 B2 U B2 F2 L D U R' D B' F2 U L' R' U2 Rw'\nF2 D2 B L2 B' R2 U2 F2 L2 U' L F' R' F' R2 U2 F' R D F Rw'\nD2 R D2 B2 R' F2 L F2 R' F L B' U F L2 B2 F2 D' R' F' Rw' Uw'\nF' L2 U2 F D2 B D2 R2 F' U' F R2 D' L R U L2 D' B2 L Fw'\nD B2 U2 F2 D2 R2 D' R2 U' F2 R F2 U' L2 F L B' D' B D2 Rw2",
        "R2 D2 R2 D' L2 B2 L2 B2 L B2 R' D2 B D L2 F' R' D F2 L' Uw2\nF2 D' L2 B2 D2 R2 U' L2 D L U R2 B F2 L F U2 R' B U2 Rw' Uw\nR2 U2 R2 B2 F2 U2 R' F2 R B L' R F2 D B D2 B D2 L U Rw2 Uw'\nU F2 R2 F2 D' L2 D' B2 F2 L2 B D U2 R B F L F2 U2 F' Rw Uw2\nF' L2 F R2 U2 B L2 F L B2 F2 D' U2 F' R2 F D R D Rw' Uw\nR2 U2 R2 D2 F2 D' F2 D' R2 F2 L D2 R2 B2 R2 F' L R D' F U Rw2 Uw2\nD2 F2 D L2 R2 D B2 U' R2 D2 B' D2 L' D2 B' L' U' R' U' F2 Uw2\nR' D2 B2 L2 B2 R' B2 U' F L' U2 R2 B D2 R D2 L' U2 Rw' Uw'\nB' L2 B' U2 F2 D2 B U2 L2 B L2 D B' F2 R' B' U2 L D L2 F' Rw\nB2 L2 U2 F2 R2 F2 L2 D F' R' U2 R U L' R D' L' B F U2 Fw Uw'\nL2 U' L2 R2 B2 D' R2 D2 R' B F2 R2 D2 F' U' L' B' R D B Rw' Uw\nL2 U' R2 D' R2 U R2 U R' B2 L' U' F2 L D' L F' D F Rw2\nD2 R2 B L2 F' L2 R2 F2 U L B2 U L D' R2 U' L U2 B Rw' Uw\nR2 F2 L2 U2 F' L2 B U2 L U' L2 F2 L' B F D' L' R' F Rw2 Uw2\nF2 R2 U B2 U L2 D2 L B' D2 R D2 B' D' F' D' B R' D2 Rw' Uw",
        "D2 R' D' L B L' F' L U F2 D' F2 D2 F' B' R2 B' U2 D2\nU F2 D R2 D2 B2 U2 R2 F2 R' B' D2 L B' F L2 B2 R B2 F2 Rw'\nR2 U2 L2 B F2 R2 D2 U2 B U' F' L' F' U F' U' B' D L' U2 Rw'\nD2 L F2 L F2 U2 R2 U' R' U' B' D F U R D2 B' F2 Uw'\nB2 U2 B U2 B L2 U2 B' L B D2 L U L' F' R D U' B Rw2\nF2 L2 F2 D2 L2 B D2 L2 F' D' L2 D F' U R F2 U' L2 F D2 Fw' Uw'\nB2 R' U2 R F2 D2 L B2 R D2 B' R U' R' B U2 B' L2 B' L' Uw\nB2 R2 U' L2 B2 R2 U' F2 D' L2 U L' F R D U' B L' D B Rw Uw\nD' B2 R2 U B2 D' U B2 L' D' F L2 R' D R B' F' D B' Rw2\nB2 L R D2 R' D2 F2 R2 B2 R D L' R' F L R2 B' U F2 L2 Uw'\nR2 F2 D2 B' R2 U2 L2 D2 U2 F2 D' R U' B' D2 U' B2 D' F' D2 U Rw\nB R2 U2 B2 L2 D2 U2 F2 L2 F' D2 R' B L B2 D2 B' D' R2 B' D2 Rw' Uw'\nF' D2 F2 D2 F D2 L2 U2 F L' D F R' F2 U L2 F2 L' D F Rw2 Uw\nB2 D U2 B2 R2 U' R2 D' L2 U F2 L' D' R2 D' B' U L F R2 B Uw2\nD F2 D U R2 U' B2 U' F2 L' D' L2 U B' D' F' R' U' F D2 Fw' Uw'"
      ],
      "scrambler": "333ni",
      "copies": 1,
      "title": "3x3x3: Multiple Blindfolded Round 2 Group B",
      "fmc": false,
      "group": "B",
      "event": "333mbf",
      "round": 2
    }
  ],
  "generationDate": "Sep 8, 2017 8:41:18 PM",
  "competitionName": "Jfly mbf test",
  "generationUrl": "http://localhost:2014/scramble/#competitionName=Jfly%20mbf%20test&rounds=i('eventID'-'333mbf'_'round'-'1'_'groupCount'-2_'scrambleCount'-2_'extraScrambleCount'-10_'copies'-1)_('eventID'-'333mbf'_'round'-'2'_'groupCount'-2_'scrambleCount'-3_'extraScrambleCount'-15_'copies'-1)!&version=1.0",
  "version": "net.gnehzr.tnoodle.server.webscrambles.ScrambleRequest-devel"
}
```

</details>

### CODE REVIEW

**Comment by Laura-O** [comment]
Tested everything and it works fine. :+1: 

I don't understand why you set the default number of scrambles to 60 though. On the one hand, it's not divisible by 7, so the half of the 9th page is just empty. On the other hand, that's indeed a crazy number. I doubt that we will see someone attempting 60 cubes in the near future and even if we do, there is still the option to adjust the number of scrambles.
I think that's mainly a waste of paper because delegates just take this as a standard and do not change it. So I would like to suggest to set the default to 28.

**Comment by jfly** [comment]
Ok, setting the default back to 28 sounds good to me!

File: scrambler-interface/WebContent/scrambler-interface/js/ui.js [hunk 1]
```diff
@@ -11,6 +11,8 @@ var mark2 = {};
 (function() {
     "use strict";
 
+    var THREE_MULTI_MAX_PUZZLES_PER_ATTEMPT = 28;
```
Comment (viroulep) [inline scrambler-interface/WebContent/scrambler-interface/js/ui.js]: nit: this sounds like 28 is the maximum number of scrambles we can generate per attempt, I'd probably change "MAX" to "DEFAULT" or something like that.

Comment (jfly) [reply scrambler-interface/WebContent/scrambler-interface/js/ui.js]: Ah, very good point! I've changed it to `THREE_MULTI_DEFAULT_PUZZLES_PER_ATTEMPT`.

**Pull request merged**